lstchain/reco/utils.py

```diff
@@ -451,13 +454,23 @@ def filter_events(events,
 
     for k in filters.keys():
         if k in events.columns:
-            filter = filter & (events[k] >= filters[k][0]) & (events[k] <= filters[k][1])
+            filter &= (events[k] >= filters[k][0]) & (events[k] <= filters[k][1])
 
-    if dropna:
+    _finite_params = list(set(finite_params + events.columns))
```

Comment:
I think you got a logic error here. You still compute `events.isnull` on all colums. and there is no `any(axis=1)` her.

Also `_finit_params` now contains all columns in events, right? I think the correct solutino would just be:

```
nans = events[finite_params].isnull()
filter &= ~nans.any(axis=1)  # not any, so all have to be finite
```
```

Reply:
damn, I tried to be too quick, thanks.